app-frontend/src/app/components/pages/project/analyses/visualize/index.html

```diff
@@ -2,9 +2,12 @@
     <div class="page-header wide">
         <h3>Key metrics</h3>
     </div>
-    <div class="page-card analysis-maps-card">
+    <div class="page-card analysis-histogram-card text-center" ng-if="$ctrl.isLoadingAnalysis">
+      <h3>Loading histograms and maps<i class="icon-load animate-spin"></i></h3>
+    </div>
+    <div class="page-card analysis-maps-card" ng-if="!$ctrl.isLoadingAnalysis">
       <div class="row">
-        <div class="column-4 flex-display analysis-viz-container" ng-repeat="analysis in $ctrl.analyses track by $index">
+        <div class="column-4 flex-display" ng-repeat="analysis in $ctrl.analyses track by analysis.id">
```

Comment:
it's occurred to me that people might be silly and list the same analysis twice in the url, and this will silently throw an error in the console (and I just did it while testing because it's easy to copy/paste).
The primary reason that we avoid using $index is because pretty much every framework does some optimization in re-using dom nodes in lists when things are re-ordered, but it can only do this optimization when things are uniquely identified. We aren't planning to allow re-ordering stuff here, so we might as well just use $index.

Reply:
@Lknechtli I added a `trackId` field (which is generated on the fly when looping on the requested analyses) to each `analysis` object so that analyses could be better tracked by it not only in here, but also in map display and histogram display (since map layers and tile histograms need unique IDs to render correctly).